dotCMS/src/main/java/com/dotcms/api/system/event/local/MethodEventSubscriber.java

```diff
@@ -0,0 +1,54 @@
+package com.dotcms.api.system.event.local;
+
+import com.dotmarketing.exception.DotRuntimeException;
+
+import java.lang.reflect.InvocationTargetException;
+import java.lang.reflect.Method;
+
+/**
+ * This implemention wraps a method in order to be called on notify.
+ * @author jsanca
+ */
+public class MethodEventSubscriber implements EventSubscriber {
+
+    private final Method method;
+    private final Object subcriber;
+    private final String id;
+
+    public MethodEventSubscriber(final Method method,
+                                 final Object subcriber) {
+
+        this.method    = method;
+        this.subcriber = subcriber;
+        this.id        = subcriber.getClass().getName() + "#"
+                                + method.getName();
+    }
+
+    @Override
+    public String getId() {
+
+        return this.id;
+    }
+
+    @Override
+    public void notify(final Object event) {
+
+        try {
+
+            this.method.invoke(this.subcriber, new Object[] { event });
+        } catch (IllegalArgumentException e) {
+
+            throw new DotRuntimeException("Invalid Argument on method: " + this.id + ", for event: " + event, e);
```

Comment:
![CRITICAL](https://sonarsource.github.io/sonar-github/severity-critical.png 'Severity: CRITICAL') Define a constant instead of duplicating this literal ", for event: " 4 times. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1192)